### PR TITLE
run tests in CI

### PR DESCRIPTION
**Pull request by drewbanin**
### Description

This PR runs automated tests in PRs via a github workflow

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have generated docs locally, and this change appears to resolve the stated issue
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Automated tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
```
Comment (leahwicz) [inline .github/workflows/test.yml:6]: Nit: I would scope down the GitHub token to only have limited read permissions

```
permissions:
  contents: read
```

**Review by leahwicz — APPROVED** [review]
LGTM!

File: src/app/services/selector_matcher.test.js [hunk 2]
```diff
@@ -442,19 +442,19 @@ test("Test getting nodes by test type", () => {
     expect(
         matchByTestType('data')
     ).toStrictEqual(
-        [7]
+        []
```
Comment (drewbanin) [inline src/app/services/selector_matcher.test.js:445]: @leahwicz could you (or someone from the Core team) confirm that the changes to these tests are correct? I _think_ that these test were invalid and, if I'm reading this correctly, the node with id=7 is a generic/schema, not a singular/data test.

Comment (leahwicz) [reply src/app/services/selector_matcher.test.js:445]: @emmyoop could you be a second pair of eyes here please?

**Pull request merged**